frontend/test/account-send.test.ts

```diff
@@ -0,0 +1,794 @@
+      checkAbstract: jest.fn(),
+      checkSafeUint53: jest.fn(),
+      throwError: jest.fn() as never,
+      throwArgumentError: jest.fn() as never,
+    };
+  });
+
+  it.each([randomInt(2, 10), randomInt(2, 10), randomInt(2, 10), randomInt(2, 10)])(
+    "End to end integration test when number '%s'",
+    async (num) => {
+      const sends = createAccountSend(num);
+
+      // Offset the account sends based on when local storage was cleared
+      let offset = 0;
+      for (const [idx, value] of sends.accountSends.entries()) {
+        const {
+          amount,
+          tokenAddress,
+          txHash,
+          recipientAddress: randomRecipientAddress,
+          advancedMode,
+          usePublicKeyChecked,
+          pubKey,
+        } = value;
+
+        // The probability of either event happening and thus clearing the local
+        // storage is about 23%.
+        const resetLocalStorageCount = Math.random() < 0.125;
+        const resetLocalStorageValues = Math.random() < 0.125;
+
+        const storeSendArgs = {
+          chainId: 5,
+          viewingPrivateKey,
+          unencryptedAccountSendData: {
+            amount: amount,
+            tokenAddress,
+            txHash,
+            // Sender address needs to be static because it is part of the key used
+            // to fetch data from localforage.
+            senderAddress: recipientAddress,
+          },
+          accountDataToEncrypt: {
+            recipientAddress: randomRecipientAddress,
+            advancedMode,
+            usePublicKeyChecked,
+            pubKey,
+          },
+        };
+
+        await storeSend(storeSendArgs);
+
+        if (resetLocalStorageCount) {
+          offset = idx;
+          await localforage.setItem(localStorageCountKey, undefined);
+        }
+
+        if (resetLocalStorageValues) {
+          offset = idx;
+          await localforage.setItem(localStorageValueKey, undefined);
+        }
+      }
+      const expectedArray = sends.accountSends
+        // We must add 1 because slice is inclusive
+        .slice(offset !== 0 ? offset + 1 : 0)
+        .map((value) => {
+          return {
+            ...value,
+            pubKey: `0x99${value.pubKey.slice(4, 26)}`,
+          };
+        })
+        .reverse();
+
+      const accountSends = await fetchAccountSends({
+        address: recipientAddress,
+        viewingPrivateKey,
+        chainId: 5,
+      });
+
+      // Remove dateSent, it will cause the tests to fail because we cannot know
+      // the exact datetime of a new Date() call.
+      const n = accountSends.map((value) => {
+        const { dateSent: _, ...rest } = value;
+        return {
+          ...rest,
+        };
+      });
+
+      // This array's order matches the expected results from the input array.We
+      // need to reverse this array because fetchAccountSends will do the same to
+      // show the most recent send first.
+      expect(n).toEqual(expectedArray);
+    },
+    15000
```

Comment:
What is this 15k value?

Reply:
For posterity, it is an override on the timeout to be 15,000ms.